### PR TITLE
Add support for more docbook elements

### PR DESCRIPTION
**Pull request by Hextremist**
Now processes:
- funcsynopsis
- function
- parameter
- quote
- sect1 .. sect5

### CODE REVIEW

**Comment by LightGuard** [comment]
Any reason not to merge this?

**Comment by mojavelinux** [comment]
I've just been super busy, so I haven't had a chance to review it yet. I'll try to get to it asap.

**Comment by mojavelinux** [comment]
Can you provide a sample document that you are converting so I can test these changes?

**Comment by mojavelinux** [comment]
It would also help if we split this pull request so that each change can be reviewed individually. Please split it based on elements that are independent from one another.

**Comment by mojavelinux** [comment]
In particular, I'd like to see the sect1, sect2, etc support split off as a separate pull request.

**Comment by mojavelinux** [comment]
I went ahead and pushed the commit for the sectN support.

**Comment by mojavelinux** [comment]
I'm happy to say that we now have a proper gem structure complete with a test suite. To expedite code merges, I'd like to ask that you create a test for each new DocBook element being converted. Use the existing tests as starting points.

I also made a number of enhancements today that addressed portions of this PR.

**Comment by Hextremist** [comment]
OK, is it is OK to make a single commit for function, parameter, and quote?  These are very basic changes to the same single line in the code, so it seems a bit overkill to make them separate commits...

**Comment by mojavelinux** [comment]
Yes, though I think I already made those changes in yesterday's
refactoring. If I missed some, don't hesitate to fill in what's missing.

**Comment by mojavelinux** [comment]
The funcsynopsis stuff is all related and can come in together, esp now
that we have tests (yeah!)

**Comment by Hextremist** [comment]
OK, I will fix this when I have some spare time.  ETA one week....

**Comment by mojavelinux** [comment]
:+1:

**Comment by Hextremist** [comment]
Finished, have a look ...

File: bin/docbookrx [hunk 1]
```diff
@@ -1,13 +1,13 @@
 #!/usr/bin/env ruby
 
-if File.exist?(app_lib_path = File.join(File.dirname(__FILE__), '../lib/docbookrx'))
+
+if File.exist?(app_lib_path = File.expand_path(File.join(File.dirname(__FILE__), '../lib/docbookrx')))
```
Comment (mojavelinux) [inline bin/docbookrx]: I think this line has to be the way we had it based on the research/testing that @mogztter did for Asciidoctor core. See https://github.com/asciidoctor/asciidoctor/blob/master/bin/asciidoctor#L5-L9. Or perhaps expand_path is harmless. Is there are reason you are expanding the path?

**Comment by mojavelinux** [comment]
Nice! I'll review asap!

File: bin/docbookrx (hunk 1 above)
Comment (Hextremist) [inline bin/docbookrx]: Like I said in the commit message, I got an exception when trying to run the script (not installed), directly from the root source directory:

/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require':
     cannot load such file -- bin/../lib/docbookrx (LoadError)
        from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
        from bin/docbookrx:4:in `<main>'

By forcing the path to be absolute (no /../), it works fine.

Comment (Hextremist) [inline bin/docbookrx]: I tried the same thing for asciidoctor, and I get the same exception:

kjell@pc asciidoctor]$ bin/asciidoctor
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- bin/../lib/asciidoctor (LoadError)
	from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
	from bin/asciidoctor:6:in `<main>'
[kjell@pc asciidoctor]$ 

This is not important tom me that you include this change, so just leave it out if you like.

Comment (mojavelinux) [inline bin/docbookrx]: Aha! It appears that it depends how you invoke it. If you add a leading `./`, then it works.

    $ ./bin/asciidoctor

I always run scripts with `./` at the beginning, which is why I never noticed it. However, I suppose it should work either way (if you agree). If so, then we'll update it everywhere.

Comment (Hextremist) [inline bin/docbookrx]: Yup, the shell does not require ./, so neither should the script  :-)

Comment (mojavelinux) [inline bin/docbookrx]: Here's the preferred way to write this line that works in both scenarios:

```ruby
if File.exist?(app_lib_path = (File.expand_path '../../lib/docbookrx', __FILE__))
```

Could you use that syntax?

Thanks for pointing this out. Now I get why others have been having trouble executing the scripts out of the development directory.

Comment (Hextremist) [inline bin/docbookrx]: (y)

**Comment by mojavelinux** [comment]
Amazing work! Thank you! I squashed and merged as 2ce2654918e2ebb38941f6170481c48a343c8e0b.

I deferred the check for a valid xmldoc because I'd like to focus on that in a separate pull request.

**Comment by mojavelinux** [comment]
Note that I also did some consolidate on the code to bring it inline with the (undocumented) coding styling that I use.

**Pull request closed without merging**